### PR TITLE
fix logit-to-multi-hot conversion in example

### LINKED ISSUE #26830: Possible error in text classification example code for multi-label classification
### System Info

N/A *(system config independent)*

### Who can help?

@ArthurZucker and @younesbelkada 

### Information

- [X] The official example scripts
- [ ] My own modified scripts

### Tasks

- [X] An officially supported task in the `examples` folder (such as GLUE/SQuAD, ...)
- [ ] My own task or dataset (give details below)

### Reproduction

N/A

### Expected behavior

Shouldn't the default condition be `p > 0` in the following lines for multi-label classification? The sequence classification model returns the logits and sigmoid has not been applied anywhere.

https://github.com/huggingface/transformers/blob/5c081e29930466ecf9a478727039d980131076d9/examples/pytorch/text-classification/run_classification.py#L658

https://github.com/huggingface/transformers/blob/5c081e29930466ecf9a478727039d980131076d9/examples/pytorch/text-classification/run_classification.py#L724

### PR DESCRIPTION
**Pull request by ranchlai**
# What does this PR do?

This PR fixes #26830 . The bug might lead to difference in measuring multi-label acc (Slightly higher precision and lower recall). 

@younesbelkada Would you please review.

### CODE REVIEW

**Review by younesbelkada — APPROVED** [review]
Thanks!

**Review by ArthurZucker — COMMENTED** [review]
Should we not apply sigmoid ? 
Otherwise fine with me. There is a way to pass a specific function to the trainer: `preprocess_logits_for_metrics` which can apply the sigmoid otherwise (but not needed here)

**Review by ArthurZucker — APPROVED** [review]

**Comment by ranchlai** [comment]
@ArthurZucker I've updated the comment for educational purpose.

**Review by ArthurZucker — APPROVED** [review]
Cool thanks!
